src/icp_canister_backend/src/lib.rs

```diff
@@ -175,6 +190,63 @@ pub fn init(token_id: Principal) {
     TOKEN_ID.with(|cell| {
         cell.borrow_mut().set(token_id).ok();
     });
+
+    let current_time = ic_cdk::api::time() / 1_000_000_000;
+    LAST_TIME_UPDATED.with(|cell| {
+        cell.borrow_mut().set(current_time).ok();
+    });
+
+    setup_episode_timer();
+}
+
+fn process_episodes() {
+    let current_episode = get_current_episode();
+    let last_processed_episode = get_last_processed_episode();
+
+    let mut total_assets_to_subtract = Nat::from(0u64);
+    let mut total_shares_to_subtract = Nat::from(0u64);
+
+    EPISODES.with(|episodes| {
+        let episodes_ref = episodes.borrow();
+
+        for episode_id in (last_processed_episode)..current_episode {
```

Comment:
last_processed_episode is already processed, it shouldn't be processed here again.